### DISCUSSION TITLE
Origami expression language to create, access, and transform trees and other data

### DISCUSSION THREAD

**Discussion by JanMiksovsky** [peggyjs/peggy#444] [Show and tell]
I posted an [apprecation of Peggy](https://jan.miksovsky.com/posts/2023/11-14-appreciations) on Mastodon, and some suggested using the "Show and tell" section of GitHub Discussions to post such appreciations directly in a project.

For the past few years I've been developing a project to make it easier to create software artifacts like websites by conceptualizing them as trees. This includes an [Origami expression language](https://graphorigami.org/language/) designed for concisely declaring the operations that convert an existing set of content (images, data, text) into the resources that make up a website. The project includes a [tutorial](https://graphorigami.org/concepts/tutorial) that walks through the creation of a simple site using this Origami language.

I recently replaced the handwritten parser combinator for that language with a parser generated by Peggy. I was delighted that this migration went so smoothly and produced a significantly more performant parser.

Many thanks to @hildjj and all the project contributors for their work on Peggy!

**Comment by hildjj** [comment]
Let us know if there's anything we can do to help.  Quickest way is usually on Discord.  You can find an invite link on https://peggyjs.org/

**Comment by reverofevil** [comment]
You're very welcome to ask any kind of questions in Discord.

Also your project reminded me of [Ur/Web](http://www.impredicative.com/ur/). If you've never seen it, I highly recommend taking a look there.

**Reply by JanMiksovsky** [reply]
Thanks for the pointer! Will check it out.
